### DISCUSSION TITLE
Lazy to support non defaults

### DISCUSSION THREAD

**Discussion by darklight9811** [solidjs/solid#1030] [Ideas]
Solidjs lazy method forces a default return from the import, maybe support default and if not found, fallback to export "root".

Something like:
``` typescript
async function lazy (cb) {
  const r = await cb()

  return createComponent(r?.default || r)
}
```
I didn't have time to read the actual lazy implementation, is just to illustrate what I mean. But why? I'm using solid-icons and I have to construct an object just to pass it back by default, like this:

``` typescript
import { lazy } from "solid-js"

const icons = {
	spinner: lazy(() => import("solid-icons/im").then(i => ({default: i.ImSpinner9}))),
}

export default icons
```

**Comment by NuroDev** [comment]
Came here to share the exact same thing after going through [the lazy loader source](https://github.com/solidjs/solid/blob/main/packages/solid/src/render/component.ts#L250-L289). By no means a priority fix needed but would be a nice QoL improvement to make allow for lazy loading tree-shaken imports and to not have to remap it to `default`.

## Current
```tsx
// App.tsx
import { lazy } from "solid-js"

const MyLazyComponent = lazy(() => import('./MyLazyComponent').then(({ MyLazyComponent }) => ({
	default: MyLazyComponent
})))
```

## Improved
```tsx
// App.tsx
import { lazy } from "solid-js"

const MyLazyComponent = lazy(() => import('./MyLazyComponent').then(({ MyLazyComponent }) => MyLazyComponent))
```
